Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -109,5 +100,49 @@ public QuestionRepository(TrappistDbContext dbContext)
             });
             return languageNameList;
         }
+
+        public async Task UpdateCodeSnippetQuestionAsync(int questionId, QuestionAC questionAC, string userId)
+        {
+            var updatedQuestion = Mapper.Map<QuestionDetailAC, Question>(questionAC.Question);
+            var updatedCodeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC, CodeSnippetQuestion>(questionAC.CodeSnippetQuestion);
+
+            if (await _dbContext.Question.Where(x => x.Id == questionId).AnyAsync())
+            {
+                updatedQuestion.Id = questionId;
+                _dbContext.Question.Update(updatedQuestion);
+                await _dbContext.SaveChangesAsync();
+
+                updatedCodeSnippetQuestion.Question = updatedQuestion;
+                _dbContext.CodeSnippetQuestion.Update(updatedCodeSnippetQuestion);
+                await _dbContext.SaveChangesAsync();
+
+                //Handling many to many relationship entity
+                //Remove all the mapping between CodeSnippetQuestion and CodingLanguage
+                var mappingToRemove = await _dbContext.QuestionLanguageMapping.Where(x => x.QuestionId == updatedCodeSnippetQuestion.Id).ToListAsync();
+                _dbContext.QuestionLanguageMapping.RemoveRange(mappingToRemove);
+
+                var updatedLanguageList = questionAC.CodeSnippetQuestion.LanguageList;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "updatedLanguageList" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)